sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIO.java

```diff
@@ -224,22 +569,13 @@ public PDone expand(PCollection<Mutation> input) {
       return PDone.in(input.getPipeline());
     }
 
+
     @Override
     public void populateDisplayData(DisplayData.Builder builder) {
       super.populateDisplayData(builder);
-      builder
-          .addIfNotNull(DisplayData.item("projectId", getProjectId()).withLabel("Output Project"))
-          .addIfNotNull(
-              DisplayData.item("instanceId", getInstanceId()).withLabel("Output Instance"))
-          .addIfNotNull(
-              DisplayData.item("databaseId", getDatabaseId()).withLabel("Output Database"))
-          .add(DisplayData.item("batchSizeBytes", getBatchSizeBytes())
-              .withLabel("Batch Size in Bytes"));
-      if (getServiceFactory() != null) {
-        builder.addIfNotNull(
-            DisplayData.item("serviceFactory", getServiceFactory().getClass().getName())
-                .withLabel("Service Factory"));
-      }
+      getSpannerConfig().populateDisplayData(builder);
+      builder.add(
+          DisplayData.item("batchSizeBytes", getBatchSizeBytes()).withLabel("Batch Size in Bytes"));
```

Comment:
Since it's not public and controllable by user, it should not be in display data